pkg/cstorvolumereplica/v1alpha1/cstorvolumereplica_test.go

```diff
@@ -0,0 +1,206 @@
+				"",
+			},
+		},
+		"Not Present 4": {
+			[]cvr{
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+				cvr{apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{}}},
+			},
+			[]string{
+				"",
+				"",
+				"",
+				"",
+			},
+		},
+	}
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			cvr := &cvrList{items: mock.cvrs}
+			if output := cvr.GetPoolUIDs(); !reflect.DeepEqual(output, mock.expectedString) {
+				t.Fatalf("test %q failed: expected %v \n got : %v \n", name, mock.expectedString, output)
+			}
+		})
+	}
+}
+
+func TestWithListObject(t *testing.T) {
+	tests := map[string]struct {
+		expectedCvrs []apis.CStorVolumeReplica
+	}{
+		"Object 1": {
+			[]apis.CStorVolumeReplica{
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object1", Labels: map[string]string{cstorPoolUIDLabelKey: "uid1"}}},
+			},
+		},
+		"Object 2": {
+			[]apis.CStorVolumeReplica{
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 2", Labels: map[string]string{cstorPoolUIDLabelKey: "uid1"}}},
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 3", Labels: map[string]string{cstorPoolUIDLabelKey: "uid2"}}},
+			},
+		},
+		"Object 3": {
+			[]apis.CStorVolumeReplica{
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 4", Labels: map[string]string{cstorPoolUIDLabelKey: "uid1"}}},
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 5", Labels: map[string]string{cstorPoolUIDLabelKey: "uid2"}}},
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 6", Labels: map[string]string{cstorPoolUIDLabelKey: "uid3"}}},
+			},
+		},
+		"Object 4": {
+			[]apis.CStorVolumeReplica{
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 7", Labels: map[string]string{cstorPoolUIDLabelKey: "uid1"}}},
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 8", Labels: map[string]string{cstorPoolUIDLabelKey: "uid2"}}},
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 9", Labels: map[string]string{cstorPoolUIDLabelKey: "uid3"}}},
+				apis.CStorVolumeReplica{ObjectMeta: metav1.ObjectMeta{Name: "object 10", Labels: map[string]string{cstorPoolUIDLabelKey: "uid4"}}},
+			},
+		},
+	}
+	for name, mock := range tests {
+		t.Run(name, func(t *testing.T) {
+			b := ListBuilder().WithListObject(&apis.CStorVolumeReplicaList{Items: mock.expectedCvrs})
+			for index, ob := range b.list.items {
+				if !reflect.DeepEqual(ob.object, mock.expectedCvrs[index]) {
+					t.Fatalf("test %q failed: expected %v \n got : %v \n", name, mock.expectedCvrs[index], ob.object)
+				}
+			}
+		})
+	}
+}
+
+func TestList(t *testing.T) {
+	tests := map[string]struct {
+		expectedCvrs []apis.CStorVolumeReplica
+	}{
+		"Object 1": {
+			[]apis.CStorVolumeReplica{
```

Comment:
Please change this table to include only the primitive data types

Reply:
done